inst/rmarkdown/templates/prepare-data-synapse-ga/skeleton/skeleton.Rmd

```diff
@@ -0,0 +1,390 @@
+
+head(public_access_default)
+```
+
+Join benefactor access table with number of files for each benefactor table to allow summarizing all files by access type.
+```{r, eval=params$eval}
+
+public_access <- merge(files_by_benefactor, public_access_default, by = "benefactorId")
+
+public_read <- public_access[, .(n_files = sum(N)), by = "READ"]
+public_read[, proportion := round(n_files / sum(n_files), digits = 3)]
+
+public_download <- public_access[, .(n_files = sum(N)), by = "DOWNLOAD"]
+public_download[, proportion := round(n_files / sum(n_files), digits = 3)]
+
+public_access_summary <- rbind(public_both[DOWNLOAD == TRUE, .(n_files, proportion)], 
+                               public_read[READ == TRUE, .(n_files, proportion)])                  
+public_access_summary[, tier := c("Files visible and downloadable", "Files visible")] # if downloadable, should always also be readable
+fwrite(public_access_summary, file = glue::glue("{data_dir}/public_access_summary.csv"))
+
+```
+
+## Google Analytics
+
+Google Analytics data needs to be prepped for generating some project-related figures that complement download stats and help point out more popular/active projects. 
+This needs a list of released projects, which was obtained from the Synapse data prep above. 
+
+### Pageviews data prep **
+
+For data access, make sure to get sufficient permissions for the Sage Google Analytics profile.
+
+First, authenticate. This will open a browser window for confirmation.
+```{r ga-auth, eval=params$eval}
+
+googleAnalyticsR::ga_auth()
+
+```
+
+You can confirm that you have access to the relevant view that we'll be querying. 
+The `propertyId` that should be present is 311611973.
+```{r, eval=params$eval}
+
+googleAnalyticsR::ga_account_list("ga4")
+```
+
+Query for project stats. This may take a few minutes.
+```{r, eval=params$eval}
+
+projects <- data_status$studyId
+date_range <- c("2022-09-01", "2023-08-31")
+
+ga_data <- usagereports::query_ga_project_stats(projects = projects, 
+                                                date_range = date_range)
+```
+
+Check out result. For example, the first result has stats broken down by page for that project.
+
+```{r, eval=params$eval}
+first(ga_data)
+```
+
+For summarization: 
+- We'll sum page views for all the pages of that project as total project views. 
+- We'll take the largest number of users seen for any project page and approximate that as maximum number of users for that project.
+
+```{r, eval=params$eval}
+
+ga_data <- rbindlist(ga_data, idcol = "projectId")
+ga_data <- ga_data[, .(sum_pageviews = sum(screenPageViews), max_users = max(totalUsers)), by = "projectId"]
+```
+
+We augment it with release groupings data. 
+We can check expected differences in views between released and non-released studies, given that non-released only have direct contributor/collaborators, while released should have additional community users.
```

Comment:
As I was reading this, one thing occurred to me. Given that metadata tables may be viewable for "under-embargo" studies, and the fact that table views are considered "downloads" in the data warehouse, both the "view" from GA and "download" metrics from the warehouse for "under-embargo" studies will be affected by this behavior.

Reply:
May just need to clarify in main report, I think we already know that "downloads" are a little inflated.